### PR TITLE
Add resolution;="optional" and cardinality:="multiple" directives bas…

### PR DESCRIPTION
**Pull request by djencks**
…ed on reference cardinality

Signed-off-by David Jencks <david_jencks@yahoo.com>

### CODE REVIEW

File: biz.aQute.bndlib/src/aQute/bnd/component/DSAnnotations.java [hunk 1]
```diff
@@ -133,11 +134,22 @@ private void addServiceCapability(String[] objectClass, Set<String> provides) {
...
+				|| cardinality == ReferenceCardinality.AT_LEAST_ONE;
 		Parameters p = new Parameters();
 		Attrs a = new Attrs();
 		a.put(Constants.FILTER_DIRECTIVE, "\"(objectClass=" + objectClass + ")\"");
 		a.put(Constants.EFFECTIVE_DIRECTIVE, "\"active\"");
+		if (optional) {
+			a.put(Constants.RESOLUTION_DIRECTIVE, "\"optional\"");
+		}
+		if (multiple) {
+			a.put("cardinality:", "\"multiple\"");
```
Comment (bjhargrave) [inline biz.aQute.bndlib/src/aQute/bnd/component/DSAnnotations.java]: We do need a constant for cardinality. Also quoting the value is good.

**Comment by bjhargrave** [comment]
Also, thus change needs test updates since the build failed.

**Comment by njbartlett** [comment]
I concurrently started working on this, as I didn't expect David to start working on it so quickly. Thanks David! I started with the tests so maybe I can merge this PR with my own.

**Comment by djencks** [comment]
neil, do you want to fix the busted tests and add the constant or should I?

**Comment by bjhargrave** [comment]
Neil is working it now

**Comment by bjhargrave** [comment]
I'll close this PR and wait for a new one from Neil. He cherry-picked your commit.

**Pull request closed without merging**